x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/common_fields/processor_type_field.tsx

```diff
@@ -68,13 +68,15 @@ export const ProcessorTypeField: FunctionComponent<Props> = ({ initialType }) =>
     <UseField<string> config={typeConfig} defaultValue={initialType} path="type">
       {(typeField) => {
         let selectedOptions: ProcessorTypeAndLabel[];
+        let helpText: string = '';
         if (typeField.value?.length) {
           const type = typeField.value;
           const descriptor = getProcessorDescriptor(type);
           selectedOptions = descriptor
             ? [{ label: descriptor.label, value: type }]
             : // If there is no label for this processor type, just use the type as the label
               [{ label: type, value: type }];
+          helpText = descriptor?.helpText || '';
```

Comment:
We could probably restructure this code now to something like:

```ts
const procesorDescriptor = getProcessorDescriptor(type);
if (procesorDescriptor) {
  selectedOptions = [{ label: procesorDescriptor.label, value: type }];
  helpText = procesorDescriptor.description; // assuming we are using the key `description`
} else {
  // If there is no label for this processor type, just use the type as the label
  selectedOptions = [{ label: type, value: type }];
}
```

I think this would read a bit easier :)